### PR TITLE
Consolidate signing code

### PR DESCRIPTION
**Pull request by oyvindronningstad**
Based on #43.

This collects and brings here code that was previously implemented in each board

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -219,4 +219,117 @@ if (CONFIG_BUILD_WITH_TFM)
...
+
+  elseif(CONFIG_TFM_MCUBOOT_IMAGE_NUMBER STREQUAL "1")
+    set_property(GLOBAL APPEND PROPERTY extra_post_build_commands
+      COMMAND ${PYTHON_EXECUTABLE} ${ZEPHYR_BASE}/scripts/mergehex.py
+        -o ${CMAKE_BINARY_DIR}/tfm_s_zephyr_ns.hex
+        $<TARGET_PROPERTY:tfm,TFM_S_HEX_FILE>
+        ${CMAKE_BINARY_DIR}/zephyr/${KERNEL_HEX_NAME}
+
+      #Sign concatenated binary image with default public key in mcuboot folder
+      COMMAND ${PYTHON_EXECUTABLE} ${TFM_MCUBOOT_DIR}/scripts/wrapper/wrapper.py
```
Comment (hakonfam) [inline CMakeLists.txt:265]: Have you tried extracting this to a function? It looks like several of the arguments passed to `wrapper.py` are the same across the three invocations.

File: CMakeLists.txt [hunk 2]
```diff
@@ -219,4 +219,117 @@ if (CONFIG_BUILD_WITH_TFM)
...
+        -k ${CONFIG_TFM_KEY_FILE_NS}
+        --public-key-format ${TFM_PUBLIC_KEY_FORMAT}
+        --align 1
+        -v ${TFM_IMAGE_VERSION_NS}
+        --pad
+        --pad-header
+        ${ADD_NS_IMAGE_MIN_VER}
+        -s auto
+        -H ${CONFIG_ROM_START_OFFSET}
+        ${CMAKE_BINARY_DIR}/tfm_s_zephyr_ns.hex
```
Comment (hakonfam) [inline CMakeLists.txt:276]: Aren't these typically stored in ${ZEPHYR_BINARY_DIR}? Disregard this comment if you are just sticking to what was done before refactoring

File: CMakeLists.txt [hunk 3]
```diff
@@ -219,4 +219,117 @@ if (CONFIG_BUILD_WITH_TFM)
...
+        ${CMAKE_BINARY_DIR}/tfm_s_zephyr_ns_signed.hex
+
+      COMMAND ${PYTHON_EXECUTABLE} ${ZEPHYR_BASE}/scripts/mergehex.py
+        -o ${CMAKE_BINARY_DIR}/tfm_merged.hex
+        $<TARGET_PROPERTY:tfm,BL2_HEX_FILE>
+        ${CMAKE_BINARY_DIR}/tfm_s_zephyr_ns_signed.hex
+    )
+
+    set_property(GLOBAL APPEND PROPERTY extra_post_build_byproducts
+      ${CMAKE_BINARY_DIR}/tfm_s_zephyr_ns.hex
```
Comment (hakonfam) [inline CMakeLists.txt:286]: Could you make variables for these files? They are referenced multiple times, and errors related to incorrect usage of these are typically hard to debug.

**Review by hakonfam — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 3 above)
Comment (oyvindronningstad) [reply CMakeLists.txt]: Done

File: CMakeLists.txt (hunk 2 above)
Comment (oyvindronningstad) [reply CMakeLists.txt]: This is what was done before.

File: CMakeLists.txt (hunk 1 above)
Comment (oyvindronningstad) [reply CMakeLists.txt]: Done

**Comment by oyvindronningstad** [comment]
CMakeLists.txt has been moved into zephyr.

**Pull request closed without merging**